test/new-e2e/shared-components/configcheck_test.go

```diff
@@ -0,0 +1,196 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package testinfradefinition
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/e2e"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+)
+
+type agentSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+}
+
+func TestAgentConfigCheckSuite(t *testing.T) {
+	e2e.Run(t, &agentSuite{}, e2e.AgentStackDef(nil))
+}
+
+type CheckConfigOutput struct {
+	CheckName  string
+	Filepath   string
+	InstanceId string
+	Settings   string
+}
+
+func MatchCheckToTemplate(checkname, input string) (*CheckConfigOutput, error) {
+	regexTemplate := fmt.Sprintf("=== %s check ===\n", checkname) +
+		"Configuration provider: file\n" +
+		"Configuration source: (?P<filepath>.*)\n" +
+		"Config for instance ID: (?P<instance>.*)\n" +
+		"(?P<settings>(?:[^~]*\n)+)" + // non-capturing group to get all settings
+		"~\n" +
+		"==="
+	re := regexp.MustCompile(regexTemplate)
+	matches := re.FindStringSubmatch(input)
+
+	// without a match, SubexpIndex lookups panic with range errors
+	if len(matches) == 0 {
+		return nil, fmt.Errorf("regexp: no matches for %s check", checkname)
+	}
+
+	filepathIndex := re.SubexpIndex("filepath")
+	instanceIndex := re.SubexpIndex("instance")
+	settingsIndex := re.SubexpIndex("settings")
+
+	return &CheckConfigOutput{
+		CheckName:  checkname,
+		Filepath:   matches[filepathIndex],
+		InstanceId: matches[instanceIndex],
+		Settings:   fmt.Sprintf("%s", matches[settingsIndex:]), // format to string for assertion
```

Comment:
**❓ question:** ‏I'm not sure to understand why `[settingsIndex:]` and not `[settingIndex]`. Made some tests locally and `matches[settingsIndex]` returns one string containing all captured data. What I am missing ?

Reply:
👍 this is a leftover from a previous iteration that had multiple captures, then I realized the output also included yaml lists so I modified it but never updated this - will fix

Reply:
fixed: https://github.com/DataDog/datadog-agent/pull/18290/commits/6b5fa1dafce6aae97b241fcc4aa31b6d47e0e092